scripts/lis_hotfix_pipeline/lis-build-vm-power-on-off.ps1

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+<#
+.Description
+    Changes the power state of LIS build VMs.
+#>
+
+param(
+    # LIS RPM Build VM's resource group.
+    [string] $ResourceGroupName,
+
+    # Subscription secret file
+    [string] $secretsFile,
+
+    # On/Off
+    [string] $Operation
+)
+
+try {
+    Invoke-WebRequest -Uri https://raw.githubusercontent.com/LIS/LISAv2/master/Libraries/TestLogs.psm1 -UseBasicParsing -OutFile ".\TestLogs.psm1"
+    Invoke-WebRequest -Uri https://raw.githubusercontent.com/LIS/LISAv2/master/Libraries/TestHelpers.psm1 -UseBasicParsing -OutFile ".\TestHelpers.psm1"
+    Import-Module ".\scripts\lis_hotfix_pipeline\Library.psm1" -Force
+    Import-Module ".\TestLogs.psm1" -Force
+    Import-Module ".\TestHelpers.psm1" -Force
+
+    Set-Variable -Name LogDir -Value ".\" -Scope Global
+    Set-Variable -Name LogFileName -Value "lis-build-vm-power-on-off.log.txt" -Scope Global
+
+    Register-AzureSubscription($secretsFile)
+
+    $VMs = Get-AzVm | Where-Object {$_.Name -inotmatch 'controller_vm' -and $_.ResourceGroupName -eq $ResourceGroupName} -Verbose
+
+    # Perform the Power ON/OFF operation
+    $Jobs = @()
+    foreach ($VM in $VMs) {
+        if ($Operation -imatch "off") {
+            Write-LogInfo "Stopping $($VM.Name)..."
+            $Jobs += $VM | Stop-AzVm -AsJob -Force
+        } elseif ($Operation -imatch "On") {
+            Write-LogInfo "Starting $($VM.Name)..."
+            $Jobs += $VM | Start-AzVm -AsJob
+        } else {
+            Write-LogErr "Unsupported VM operation '$Operation'. Supported operations : On/Off"
+            exit 1
+        }
+    }
+
+    $InCompleteJobs = ($Jobs | Where-Object {$_.State -imatch "running"}).Count
+    while ($InCompleteJobs -ne 0) {
+        Write-LogInfo "'$InCompleteJobs' jobs still running. Waiting 10 seconds..."
+        Start-Sleep -Seconds 10
+        $InCompleteJobs = 0
+        foreach ($Job in $Jobs) {
+            $JobStatus = Get-Job -Id $Job.Id
+            if ($JobStatus.State -imatch "running") {
+                $InCompleteJobs += 1
+            }
+        }
+    }
+
+    # Remove the jobs once completed.
+    Write-Host "Removing $($Jobs.Count) completed background jobs..."
+    $Jobs | Remove-Job -Force
+
+    # Verify SSH for all VMs...
+    if ($Operation -imatch "On") {
+        $OfflineVMs = 0
+        $Resources = Get-AzResource | Where-Object {$_.ResourceGroupName -eq $ResourceGroupName }
+        $LB = $Resources | Where-Object { $_.ResourceType -eq "Microsoft.Network/loadBalancers" }
+        $LB = Get-AzLoadBalancer -Name $LB.Name -ResourceGroupName $ResourceGroupName
+        $NatRules = $LB.InboundNatRules | Where-Object {$_.BackendIPConfiguration.Id -ne $null }
+        $PublicIP = $Resources | Where-Object { $_.ResourceType -eq "Microsoft.Network/publicIPAddresses" }
```

Comment:
Why hardcoding ?

Reply:
Those strings are Azure defined, and required.